### PR TITLE
Features/geo area rest graphql

### PR DESCRIPTION
**Pull request by shreeyash07**
- Addresses
  - #2904 
  - #2909 
  - #2881
  - #2882 
  - #2894
  - #2906
  - #2907
  - #2908 
  
- Depends on
  - https://github.com/the-deep/server/pull/1472
  - https://github.com/the-deep/server/pull/1449

## Changes

* Rest to Graqhql for 
  - Create Region
  - Create Admin Level
  - Update Admin Level
  - Publish Region
  - Delete Region
  - Fetch Admin level
* Replace file upload with rest to graphql

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: app/components/GalleryFileUpload/index.tsx [hunk 1]
```diff
@@ -50,6 +53,10 @@ function FileUpload(props: Props) {
         disabled,
         projectIds,
         buttonOnly,
+        // value: valueFromProps,
```
Comment (barshathakuri) [inline app/components/GalleryFileUpload/index.tsx]: Remove unwanted comments.

File: app/components/GalleryFileUpload/index.tsx [hunk 2]
```diff
@@ -122,16 +132,31 @@ function FileUpload(props: Props) {
...
             className={_cs(
                 styles.galleryFileUpload,
                 buttonOnly && styles.buttonOnly,
             )}
             name={undefined}
-            value={null}
+            value={undefined}
             onChange={handleFileInputChange}
-            status={undefined}
+            status={status ? currentStatus : ''}
```
Comment (barshathakuri) [inline app/components/GalleryFileUpload/index.tsx]:
```suggestion
            status={status ? currentStatus : undefined}
```

File: app/components/GalleryFileUpload/index.tsx [hunk 3]
```diff
@@ -122,16 +132,31 @@ function FileUpload(props: Props) {
         ],
     );
 
+    const currentStatus = useMemo(() => {
+        if (isNotDefined(status)) {
+            return '';
```
Comment (barshathakuri) [inline app/components/GalleryFileUpload/index.tsx]:
```suggestion
            return undefined;
```

File: app/components/selections/RegionSelectInput/index.tsx [hunk 4]
```diff
@@ -69,6 +72,7 @@ function RegionSelectInput<K extends string, GK extends string>(
     const variables = useMemo(() => ({
         excludeProject: [projectId],
         search: debouncedSearchText,
+        public: true,
```
Comment (AdityaKhatri) [inline app/components/selections/RegionSelectInput/index.tsx:75]: Nice. Much awaited :smile:

File: app/views/ProjectEdit/GeoAreas/CustomGeoAddModal/index.tsx [hunk 5]
```diff
@@ -12,22 +14,39 @@ import {
...
+            data: $data
+        ) {
+            ok
+            errors
+            result {
+                id
+                isPublished
+            }
+
+        }
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/GeoAreas/CustomGeoAddModal/index.tsx]: fix indentation

File: app/views/ProjectEdit/GeoAreas/RegionCard/AddAdminLevelPane/AddAdminLevelForm/index.tsx [hunk 6]
```diff
@@ -22,32 +19,96 @@ import {
...
+                        file {
+                            url
+                        }
+                        id
+                        title
+                        mimeType
+                    }
+                    level
+                    id
+                }
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/GeoAreas/RegionCard/AddAdminLevelPane/AddAdminLevelForm/index.tsx]: fix indent

File: app/views/ProjectEdit/GeoAreas/RegionCard/AddAdminLevelPane/AddAdminLevelForm/index.tsx [hunk 7]
```diff
@@ -59,30 +120,44 @@ const schema: FormSchema = {
...
+    const {
+        geoShapeFile,
+        ...remainingValues
+    } = valueFromProps;
+
+    const initialFormValue: PartialAdminLevelInput = {
+        ...remainingValues,
+        geoShapeFile: geoShapeFile?.id,
+        region: regionId,
+    };
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/GeoAreas/RegionCard/AddAdminLevelPane/AddAdminLevelForm/index.tsx]: use useMemo

File: app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx [hunk 8]
```diff
@@ -87,63 +135,75 @@ function RegionCard(props: Props) {
...
+            },
         },
-    });
+    );
 
     const {
         pending: deleteAdminLevelPending,
         trigger: deleteAdminLevel,
-    } = useLazyRequest<unknown, number>({
+    } = useLazyRequest<unknown, string>({
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx]: need to replace this function as well

File: app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx [hunk 9]
```diff
@@ -154,72 +214,97 @@ function RegionCard(props: Props) {
...
                 onTempAdminLevelChange(undefined);
             }
 
             setAdminLevels((oldAdminLevels) => {
                 const newAdminLevels = [...oldAdminLevels];
                 const index = newAdminLevels.findIndex((item) => item.id === value.id);
-                const valueWithClientId = {
-                    ...value,
-                    clientId: value.id.toString(),
-                };
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx]: Are you sure we don't need this?

File: app/views/ProjectEdit/GeoAreas/RegionsMap/index.tsx [hunk 10]
```diff
@@ -63,44 +74,62 @@ function RegionsMap(props: Props) {
...
-                    body: {
-                        regions: [
-                            ...regions.map((d) => ({ id: d.id })),
-                            { id: value },
-                        ],
+            if (value) {
+                patchRegion({
+                    variables: {
+                        projectId,
+                        regionId: [String(value)],
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/GeoAreas/RegionsMap/index.tsx:128]: Won't this replace the previous regions and only set one?

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/components/GalleryFileUpload/index.tsx [hunk 11]
```diff
@@ -1,9 +1,9 @@
-import React, { useState, useCallback } from 'react';
+import React, { useState, useCallback, useMemo } from 'react';
 import { FileInput, useAlert } from '@the-deep/deep-ui';
 import { IoCloudUpload } from 'react-icons/io5';
 import { gql, useMutation } from '@apollo/client';
 import { removeNull } from '@togglecorp/toggle-form';
-import { _cs } from '@togglecorp/fujs';
+import { _cs, isDefined, isNotDefined } from '@togglecorp/fujs';
```
Comment (subinasr) [inline app/components/GalleryFileUpload/index.tsx]: Let's separate these out in multiple lines.

File: app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx (hunk 8 above)
Comment (shreeyash07) [reply app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx]: Backend work is left.

File: app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx (hunk 9 above)
Comment (shreeyash07) [reply app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx]: yes i checked it, we don't need it anymore

File: app/views/ProjectEdit/GeoAreas/RegionsMap/index.tsx (hunk 10 above)
Comment (shreeyash07) [reply app/views/ProjectEdit/GeoAreas/RegionsMap/index.tsx:128]: Mutation only requires selected region for the patch not the entire list.

File: app/views/ProjectEdit/GeoAreas/RegionCard/AddAdminLevelPane/AddAdminLevelForm/index.tsx [hunk 12]
```diff
@@ -92,52 +187,151 @@ function AddAdminLevelForm(props: Props) {
...
+                } = response.updateAdminLevel;
+
+                if (errors) {
+                    const formError = transformToFormError(removeNull(errors) as ObjectError[]);
+                    setError(formError);
+                }
+
+                if (isDefined(result) && ok) {
+                    alert.show(
+                        'Admin level is successfully update!',
```
Comment (subinasr) [inline app/views/ProjectEdit/GeoAreas/RegionCard/AddAdminLevelPane/AddAdminLevelForm/index.tsx]:
```suggestion
                        'Admin level is successfully updated!',
```

File: app/views/ProjectEdit/GeoAreas/RegionCard/AddAdminLevelPane/AddAdminLevelForm/index.tsx [hunk 13]
```diff
@@ -92,52 +187,151 @@ function AddAdminLevelForm(props: Props) {
...
+
+                            } as AdminLevelInputType,
+                        },
+                    });
+                }
+            },
         );
         submit();
-    }, [setError, validate, addAdminLevelTrigger]);
+    }, [setError, validate, updateAdminLevel, createAdminLevel, valueFromProps.id]);
```
Comment (subinasr) [inline app/views/ProjectEdit/GeoAreas/RegionCard/AddAdminLevelPane/AddAdminLevelForm/index.tsx]: Let's separate these in multiple lines.

File: app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx [hunk 14]
```diff
@@ -1,14 +1,14 @@
 import React, { useCallback, useMemo, useState } from 'react';
 import {
-    _cs,
-    randomString,
+    _cs, isNotDefined,
```
Comment (subinasr) [inline app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx]: Separate lines please.

File: app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx [hunk 15]
```diff
@@ -87,139 +144,198 @@ function RegionCard(props: Props) {
...
-            }));
-            setAdminLevels(adminLevelsWithClientId);
+        loading: adminLevelsPending,
+        refetch: refetchAdminLevels,
+    } = useQuery<AdminLevelsQuery, AdminLevelsQueryVariables>(
+        ADMIN_LEVELS,
+        {
+            skip: !isExpanded,
+            variables: adminLevelQueryVariables,
+            // NOTE: this prop is required for onCompleted to be
```
Comment (subinasr) [inline app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx:204]: Nice!

File: app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx [hunk 16]
```diff
@@ -87,139 +144,198 @@ function RegionCard(props: Props) {
...
+                }
+
+                const {
+                    deletedResult,
+                    errors,
+                } = response.project.projectRegionBulk;
+
+                if (errors) {
+                    alert.show(
+                        'Failed to publish selected region.',
```
Comment (subinasr) [inline app/views/ProjectEdit/GeoAreas/RegionCard/index.tsx]:
```suggestion
                        'Failed to delete selected region.',
```

**Review by subinasr — CHANGES_REQUESTED** [review]

File: app/views/ProjectEdit/GeoAreas/RegionsMap/index.tsx (hunk 10 above)
Comment (AdityaKhatri) [reply app/views/ProjectEdit/GeoAreas/RegionsMap/index.tsx:128]: Okayy, thenks

**Pull request merged**